Test1/Test1/ThreadSafePriorityQueue.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace Test1;
+
+public class ThreadSafePriorityQueue<P, V > where P : IComparable<P>
+{
+    private readonly SortedDictionary<P, List<V>> queue = new();
+    private int count = 0;
+    private readonly object queueLock = new();
+
+    public void Enqueue(P priority, V value)
+    {
+        lock (queueLock)
+        {
+            if (queue.ContainsKey(priority) != true)
```

Comment:
```suggestion
            if (!queue.ContainsKey(priority))
```
Сравнивать два значения типа `bool` - моветон.

Reply:
За тавтологию для типа `bool`: -0.5.